src/unitxt/metrics.py

```diff
@@ -1341,6 +1340,40 @@ def compute(
         return result
 
 
+class StringContainmentRatio(InstanceMetric):
+    """Metric that returns the ratio of values from a specific field contained in the prediction.
+
+    Attributes:
+        field: The field from the task_data that contains the values to be checked for containment.
+               Example task:
+                    Task(
+                        input_fields={"question": str},
+                        reference_fields={"entities": str},
+                        prediction_type=str,
+                        metrics=["string_containment_ratio[field=entities]"],
+                    )
+    """
+
+    reduction_map = {"mean": ["string_containment"]}
+    main_score = "string_containment"
+    ci_scores = ["string_containment"]
+    field: str = "entities"
```

Comment:
Thanks. Looks good. I suggest not having a default, so people won't forget to set it by mistake.

Reply:
And add a check that is set.

Reply:
Yes, I tried it first, but I got

```
❯ python prepare/metrics/string_containment.py
Metric tested successfully!
Artifact metrics.string_containment saved to /home/marukaz/ghq/github.com/marukaz/unitxt/src/unitxt/catalog/metrics/string_containment.json
Traceback (most recent call last):
  File "/home/marukaz/ghq/github.com/marukaz/unitxt/prepare/metrics/string_containment.py", line 40, in <module>
    metric = StringContainmentRatio()
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/marukaz/ghq/github.com/marukaz/unitxt/src/unitxt/dataclass.py", line 349, in custom_cls_init
    original_init(self, *args, **kwargs)
  File "/home/marukaz/ghq/github.com/marukaz/unitxt/src/unitxt/dataclass.py", line 349, in custom_cls_init
    original_init(self, *args, **kwargs)
  File "/home/marukaz/ghq/github.com/marukaz/unitxt/src/unitxt/dataclass.py", line 349, in custom_cls_init
    original_init(self, *args, **kwargs)
  [Previous line repeated 5 more times]
  File "/home/marukaz/ghq/github.com/marukaz/unitxt/src/unitxt/dataclass.py", line 504, in __init__
    raise RequiredFieldError(
unitxt.dataclass.RequiredFieldError: Required field 'field' of class StringContainmentRatio not set in StringContainmentRatio
```

then I also tried to set a value when creating the instance like `metric = StringContainmentRatio(field=reference_field)`. In that case, the json became

```
{
    "__type__": "string_containment_ratio",
    "field": "entities"
}
```

so I thought  this became the fixed value.

Reply:
In fact, [BertScore](https://github.com/IBM/unitxt/blob/9c4002caa0e2311eb542823ca2ab43122b3202f1/src/unitxt/metrics.py#L2505) has an attribute called `model_name` that doesn't have a default value. However, the model name is specified in the [prepare](https://github.com/IBM/unitxt/blob/9c4002caa0e2311eb542823ca2ab43122b3202f1/prepare/metrics/rag.py#L6), and looking at the task example in the [documentation](https://github.com/IBM/unitxt/blob/9c4002caa0e2311eb542823ca2ab43122b3202f1/docs/docs/adding_metric.rst), it doesn't seem to be implemented in a way that makes `model_name` mandatory.

Reply:
I may be misunderstanding something as I haven't fully grasped the principles of unitxt. I'd be grateful if you could help me 🙇

Reply:
Should I set default of `field` to `None` and check that it is not `None` in `compute()`?

Reply:
Yes.  Setting it to none and checking sound right.  Please use override the verify() method.

def verify(self):
    super().verify()
    If self.field is None:
         .....

Reply:
> I may be misunderstanding something as I haven't fully grasped the principles of unitxt. I'd be grateful if you could help me 🙇

Yes.  Unitxt has some principle that are not immediate to understand.  They have their benefits in the ability to reuse and supports a large variety of verification needs, but we continuously try to simplify and better document them.

Reply:
Thank you! I updated code